### PR TITLE
linux: use GTK_USE_PORTAL=1 to support the XDG Desktop Portal

### LINKED ISSUE #50386: I set Dolphin as default file manager, but VSCode opens Nautilus instead of Dolphin
- **VSCode Version:** 1.24.0-insider
- **OS Version:** Ubuntu 18.04 with GNOME, KDE, Unity and XFCE

Steps to Reproduce:

1. I set Dolphin as default file manager via command and in the defaults.list.
2. I open VSCode Insiders
3. I open a folder
4. VSCode opens Nautilus instead of Dolphin

VSCode does not recognise that Dolphin is a new default file manager. 

Does this issue occur when all extensions are disabled? Yes

### PR DESCRIPTION
**Pull request by tristan957**
Electron 14 will have support for file managers through the XDG Desktop
Portal Specification. In order to explicitly opt into support assuming
a user has an xdg-desktop-portal-* package installed, GTK needs to be
told to use it. This will allow for alternative file managers on KDE or
wlroots-based systems.

This PR fixes #50386 amongst others, once VSCode adopts Electron v14.

### CODE REVIEW

File: resources/linux/bin/code.sh [hunk 1]
```diff
@@ -48,6 +48,11 @@ else
 	fi
 fi
 
+# if the user has GTK_USE_PORTAL defined in their evironment, respect it
```
Comment (jogo-) [inline resources/linux/bin/code.sh:51]: Typo: evironment -> e**n**vironment

**Comment by Ashark** [comment]
@tristan957 I have built the vscodium (also tried vscodium-git from aur) with your patch. And it did not work. Then I have found out how to check electron version of app (from [here](https://stackoverflow.com/a/50403565/7869636)). Surprisingly, it was 13.5.1. But I only have the `electron` package in system, which is currently v15.2.0. So it looks like vscodium brings its own electron 13.

Now the question is, is it possible to switch it to use system's version of electron?

**Review by deepak1556 — DISMISSED** [review]
LGTM following Electron 16 update to `main` branch.

**Comment by tristan957** [comment]
Should we set GTK_USE_PORTAL=1 unconditionally?

**Comment by deepak1556** [comment]
> Is using export here correct? Probably don't want to pollute the parent environment.

Yeah you are right, can you update the PR. Thanks!

Btw, this code path will get executed only user launches the application from the cli, if they are triggered from launchers then we might need to set it up in https://github.com/microsoft/vscode/blob/main/resources/linux/code.desktop

**Comment by tristan957** [comment]
I could give it a go. Need to break out my dbus-foo

**Comment by Moonlight63** [comment]
I just tried this on code-insiders build on electron 16. It didn't work.

**Comment by tristan957** [comment]
What didn't work?

**Comment by Moonlight63** [comment]
Sorry, using the gtk protal env var didn't allow vscode to use the plasma filepicker when I tried it. I have since switched back to gnome so it no longer affects me, but I wonder if anyone else has gotten it working?

**Comment by tristan957** [comment]
@Moonlight63 are you using a version of VSCode with electron >= v14 and do you have xdg-desktop-portal-kde installed?

**Comment by Moonlight63** [comment]
Yes, I did. I had vs code insiders and other applications like Firefox did respect the portal, but vs code still ignored it. Like I said, I just switched back to gnome.

**Review by sonnyp — COMMENTED** [review]
`GTK_USE_PORTAL` was a debug flag and shouldn't be used. It has been moved to a flag in `GDK_DEBUG` for clarity. 

Here is a small recap

Thanks to @tristan957 (thanks!) work in https://github.com/electron/electron/pull/19159, Electron already use the native Linux file picker if it's running in a sandboxed env (Snap / Flatpak).

IIRC in order for Electron to always use the native file picker, even when not running sandboxed, Electron needs to swap [GtkFileChooserNative](https://docs.gtk.org/gtk4/class.FileChooserNative.html) for the new [GtkFileDialog](https://docs.gtk.org/gtk4/class.FileDialog.html).

However, `GtkFileDialog` is only available in Gtk 4.10 .  I don't know what Electron policy is regarding dependencies.

Should I file an issue on the Electron repository, and we can pick it up there?

I suggest closing this PR as it won't do anything.

**Comment by deepak1556** [comment]
Thanks for summarizing @sonnyp. Agree with closing this PR, as for gtk4 support it is currently tracked under https://github.com/electron/electron/issues/33690

Additionally we would want to refactor the file dialog impl in Electron to reuse Chromium's impl, ref https://github.com/electron/electron/pull/35110#issuecomment-1199123644. I am looking into that as part of https://github.com/microsoft/vscode/issues/146422

**Pull request closed without merging**